data/modules/goja-usage.ts

```diff
@@ -0,0 +1,1684 @@
+
+module NKRuntime {
+    /**
+     * The context of the current execution; used to observe and pass on cancellation signals.
+     */
+    export type ContextKey = "env" | "execution_mode" | "node" | "query_params" | "user_id" | "username" | "vars" | "user_session_exp" | "session_id" | "client_ip" | "client_port" | "match_id" | "match_node" | "match_label" | "match_tick_rate"
+    export type Context = { [K in ContextKey]: string };
+
+    /**
+     * An RPC function definition.
+     */
+    export interface RpcFunction {
+        /**
+         * An RPC function to be executed when called by ID.
+         *
+         * @param ctx - The context for the execution.
+         * @param logger - The server logger.
+         * @param nk - The Nakama server APIs.
+         * @param payload - The input data to the function call. This is usually an escaped JSON object.
+         * @returns A response payload or error if one occurred.
+         */
+        (ctx: Context, logger: Logger, nk: Nakama, payload: string): string;
+    }
+
+    /**
+     * A Before Hook function definition.
+     */
+    export interface BeforeHookFunction {
+        /**
+         * A Register Hook function definition.
+         * @param ctx - The context for the execution.
+         * @param logger - The server logger.
+         * @param nk - The Nakama server APIs.
+         * @param payload - The input data to the function call. This is usually an escaped JSON object.
```

Comment:
I believe the description is wrong, we return a map with the unmarshalled JSON payload, so on the Javascript side this will be a plain object (in Lua we return Table I believe). Not sure what type the payload should be otherwise 🤔